controller/server/gui.ml

```diff
@@ -5,68 +5,26 @@ open Opium.App
 
 let log_src = Logs.Src.create "gui"
 
-(* Require the resource directory to be at a directory fixed to the binary location. This is not optimal, but works for the moment. TODO: figure out a better way to do this.
-*)
+(* Require the resource directory to be at a directory fixed to the binary
+ * location. This is not optimal, but works for the moment. *)
 let resource_path end_path =
   let open Fpath in
   (Sys.argv.(0) |> v |> parent) / ".." / "share" // end_path
   |> to_string
 
 
-(* Load a template file
-*)
-let template name =
-  Fpath.(resource_path (v "template" / (name ^ ".mustache")))
-  |> Util.read_from_file log_src
-  >|= Mustache.of_string
-
-(* Load a partial file
- * This had to be defined separately from the function loading templates,
- * because its has to have the type string -> Mustache.t option (cannot use lwt).
-*)
-let load_partial name =
-   let load_file f =
-     let ic = open_in f in
-     let n = in_channel_length ic in
-     let s = Bytes.create n in
-     really_input ic s 0 n;
-     close_in ic;
-     Bytes.to_string s
-   in
-   let partial =
-     Fpath.(resource_path (v "template" / "partials" / (name ^ ".mustache")))
-     |> load_file
-     |> Mustache.of_string
-   in
-   Some partial
-
-(* Helper to render template *)
-let render name dict =
-  let%lwt template = template name in
-  Mustache.render ~strict:false ~partials:load_partial template (Ezjsonm.dict dict)
-  |> return
-
 (* Middleware that makes static content available *)
 let static () =
   let static_dir = resource_path (Fpath.v "static") in
   Logs.debug (fun m -> m "static content dir: %s" static_dir);
   Opium.Middleware.static ~local_path:static_dir ~uri_prefix:"/static" ()
 
-(* Main page *)
-let index content =
-  render "index" ["content", content |> Ezjsonm.string]
-  >|= Response.of_string_body
-
-let page identifier page_values =
-  let menu_flag = ( "is_" ^ identifier, Ezjsonm.bool true ) in
-  let index_with_menu_flag content =
-    render "index" (menu_flag :: ["content", content |> Ezjsonm.string])
-  in
-  render identifier page_values
-  >>= index_with_menu_flag
-  >|= Response.of_string_body
+let page html =
+  Format.asprintf "%a" (Tyxml.Html.pp ()) html
+  |> Response.of_string_body
```

Comment:
I agree it would be less verbose, and possible because we always use _Lwt.return_ on top of _page_. But it would be wrongly qualified as an effect function. I think I still prefer the verbosity of this solution.

Reply:
:+1: